src/eos-dropbox-app

```diff
@@ -198,10 +199,49 @@ class DropboxLauncher():
             logging.info("Not the main launcher instance. Exiting")
             self._mainloop.quit()
 
+    def _launch_dropbox(self):
+        self._launch_dropbox_daemon()
+        self._setup_config_monitor()
+
+    def _setup_config_monitor(self):
+        config = Gio.File.new_for_path(os.path.expanduser(os.path.expanduser(DROPBOX_CONFIG)))
+        self._config_monitor = config.monitor(Gio.FileMonitorFlags.NONE)
+        self._config_monitor.connect('changed', self._on_config_changed)
+
     def _launch_dropbox_daemon(self):
         logging.info("Launching Dropbox's daemon at {}...".format(DROPBOX_LAUNCHER))
         subprocess.Popen([DROPBOX_LAUNCHER])
 
+    def _on_config_changed(self, monitor, file_obj, other_file, event_type):
+        logging.info("Config file monitor {}: {}".format(file_obj.get_path(), event_type))
+        if event_type != Gio.FileMonitorEvent.CHANGES_DONE_HINT:
+            if event_type == Gio.FileMonitorEvent.DELETED and self._dir_monitor:
+                self._dir_monitor.cancel()
+            return
+        if os.path.exists(file_obj.get_path()):
+            logging.info("Configuration for Dropbox created; opening folder when created...")
+            self._open_dropbox_when_created()
+
+    def _open_dropbox_when_created(self):
+        dropbox_dir = get_dropbox_directory()
+        if dropbox_dir is None:
+            logging.warning("No Dropbox folder configured yet. Cannot open or monitor it!")
+            return
+        if os.path.exists(dropbox_dir):
+            self._open_dropbox_directory()
+            return
+        logging.info("Setting up monitor for folder {}...".format(dropbox_dir))
+        dir_obj = Gio.File.new_for_path(dropbox_dir)
+        self._dir_monitor = dir_obj.monitor(Gio.FileMonitorFlags.NONE)
+        self._dir_monitor.connect('changed', self._on_dir_changed)
+
+    def _on_dir_changed(self, monitor, file_obj, other_file, event_type):
+        logging.info("Dropbox dir monitor {}: {}".format(file_obj.get_path(), event_type))
+        if event_type != Gio.FileMonitorEvent.CREATED:
+            return
+        logging.info("Dropbox folder created; opening now...")
+        self._open_dropbox_directory()
+        self._dir_monitor.cancel()
```

Comment:
Yes, it's not needed, but I think it would be nice to clean that up. No big deal, though